### ISSUE TITLE
Get WebRTC backend working again

### ISSUE THREAD

**Issue by juj** [emscripten-core/emscripten#3818]
Poking at the test `sockets.test_webrtc`, I see that we have removed the whole WebRTC backend from Emscripten altogether. WebRTC used to be implemented in `src/library.js`, see e.g. at this point of time: https://github.com/kripken/emscripten/blob/3b99bb7bef220ad98760b117d14854098f86db9e/src/library.js#L7032, where the two implementations WebRTC vs WebSockets were chosen based on a `#if SOCKET_WEBRTC`. Looks like the WebRTC part of it was never integrated as part of the filesystem in `src/library_sockfs.js`, which only carries WebSockets -specific code. 

This commit https://github.com/kripken/emscripten/commit/d92efe09c6bc825539f9ce8f7e10ac2fcde0bdaf removes the sockets specific code from `src/library.js`, but looks like the WebRTC implementation was not migrated anywhere else, so it got lost as part of the refactoring.

**Comment by kripken** [comment]
Yes, it was never moved to sockfs, and testing was broken, so I didn't have a reliable way to work on it. We could resurrect it if we figure out how to get testing working again.

**Comment by luser** [comment]
I've got a patch to fix the webrtc backend on top of master:
https://github.com/luser/emscripten/commits/fix-webrtc-sockets

It passes test_webrtc fine here with Firefox. I rebased that patch on top of incoming, but it sounds like more work will be necessary to actually get it working again.

**Comment by luser** [comment]
So AFAICT, what needs to happen is to port the code from here:
https://github.com/kripken/emscripten/blob/3b99bb7bef220ad98760b117d14854098f86db9e/src/library.js#L7032

to library_sockfs.js, in a `webrtc_sock_ops` like the existing `websocket_sock_ops`?

**Comment by kripken** [comment]
Yes, that's correct.

**Comment by kripken** [comment]
@luser's patches are at https://github.com/luser/emscripten/commits/fix-webrtc-sockets

**Comment by kripken** [comment]
Separately the patch for SDL_net in there could be landed, if we add a test.

**Comment by luser** [comment]
Sorry, I corrected myself on IRC, that was a slightly outdated branch, the working changes are here: https://github.com/luser/emscripten/commits/webrtc-sockets

**Comment by luser** [comment]
That patch is against master, it'd need a bit of rework to get it working against incoming (as mentioned above, refactoring it to work with sockfs).

**Comment by kkimdev** [comment]
Hi, what's the status of this?

**Comment by jkbbwr** [comment]
Hi was there any progress on this?

**Comment by VinnyVicious** [comment]
Can this be re-opened?
